addons/test_mail_full/tests/test_portal.py

```diff
@@ -162,3 +165,143 @@ def test_portal_mixin(self):
+            'name': 'Wood Corner',
+            'email': 'wood_corner@example.com',
+        })
+        cls.record_customer = cls.env['mail.test.portal'].create({
+            'partner_id': cls.customer.id,
+            'name': 'Test Portal Record',
+            'user_id': cls.user_admin.id,
+        })
+        cls.mail_template = cls.env['mail.template'].create({
+            'name': 'Test send quotation',
+            'subject': 'Your quotation',
+            'body_html': '<p>Hello, your quotation is ready for review.</p>',
+            'auto_delete': True,
+            'email_from': '{{ (object.user_id.email_formatted or user.email_formatted) }}',
+            'partner_to': '{{ object.partner_id.id }}',
+            'model_id': cls.env.ref('test_mail_full.model_mail_test_portal').id,
+        })
+        cls._create_portal_user()
+        # Get access URL that is embedded in the customer email for accessing the record.
+        for group_name, group_func, group_data in cls.record_customer.sudo()._notify_get_recipients_groups():
+            if group_name == 'portal_customer' and group_func(cls.customer):
+                cls.record_access_url = group_data['button_access']['url']
+                break
+        else:
+            raise AssertionError('Record access URL not found')
+        # Transform that access URL to have one with an invalid access token
+        cls.record_access_url_wrong_token = re.sub(r'access_token=[0-9a-z-]*',
+                                                   'access_token=00000000-0000-0000-0000-000000000000',
+                                                   cls.record_access_url)
+
+    def test_initial_data(self):
+        """Test some initial value."""
+        self.assertNotEqual(self.record_access_url, self.record_access_url_wrong_token)
+
+    def _get_composer_with_context(self, template_id=False):
+        return self.env['mail.compose.message'].with_context({
+            'default_model': self.record_customer._name,
+            'default_res_id': self.record_customer.id,
+            'default_use_template': template_id,
+            'default_template_id': template_id,
+            'default_composition_mode': 'comment',
+            'mark_so_as_sent': True,
+            'default_email_layout_xmlid': 'mail.mail_notification_layout_with_responsible_signature',
+            'force_email': True,
+            'model_description': 'Quotation',
+        })
+
+    @users('employee')
+    def test_send_message_to_customer_using_template(self):
+        """Send a mail to a customer without an account and check that it contains a link to view the record.
+
+        Other tests below check that that same link has the correct behavior.
+        This test follows the common use case by using a template while the next send the mail without a template."""
+        composer = self._get_composer_with_context(self.mail_template.id).create({})
+        # currently onchange necessary
```

Comment:
I know this comment :disguised_face:

Reply:
:-) I have copied it in case we want to change that in the future.